### PR TITLE
fish: 3.3.1 -> 3.4.0

### PR DESCRIPTION
**Pull request by kidonng**
###### Description of changes

https://github.com/fish-shell/fish-shell/releases/tag/3.4.0

Resubmit in light of https://github.com/NixOS/nixpkgs/pull/163886#issuecomment-1065971268

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by schuelermine** [comment]
This seems to work for me. Fish & docs are built and work

**Comment by cole-h** [comment]
From that issue, it seems that the pexpect tests are just flaky (and indeed, I was able to get a few successful builds on the community box every once in a while). Could you disable the tests that they think are most likely to be flaky (i.e. all the pexpect tests and the `history_*` tests) and leave a link back to that issue?

**Comment by kidonng** [comment]
> From that issue, it seems that the pexpect tests are just flaky (and indeed, I was able to get a few successful builds on the community box every once in a while).

Exactly what I thought, the test error doesn't look that serious and I assume it's not due to fish itself.

> Would you please disable the tests that they think are most likely to be flaky (i.e. all the pexpect tests and the `history_*` tests) and leave a link back to that issue? 🙏

I'm not sure, but seems like it's just `tests/pexpects/exit_handlers.py` on aarch64-linux that's on problematic. We can disable more if needed.

**Comment by schuelermine** [comment]
Is `rm`ing the tests the best way to disable them? Apparently there is an environment variable that disables some of them ($CI)

**Comment by cole-h** [comment]
> > Would you please disable the tests that they think are most likely to be flaky (i.e. all the pexpect tests and the `history_*` tests) and leave a link back to that issue? pray
> 
> I'm not sure, but seems like it's just `tests/pexpects/exit_handlers.py` on aarch64-linux that's on problematic. We can disable more if needed.

You're right, a smaller scope is better. Thanks!

File: pkgs/shells/fish/default.nix [hunk 1]
```diff
@@ -144,11 +144,9 @@ let
       # --version`), as well as the local documentation for all builtins (and
       # maybe other things).
       url = "https://github.com/fish-shell/fish-shell/releases/download/${version}/${pname}-${version}.tar.xz";
-      sha256 = "sha256-tbTuGlJpdiy76ZOkvWUH5nXkEAzpu+hCFKXusrGfrok=";
+      sha256 = "sha256-tbSKuEhrGe9xajL39GuIuepTVhVfDpZ+6Z9Ak2RUE8U=";
```
Comment (cole-h) [inline pkgs/shells/fish/default.nix:147]: Interesting how the hashes start with the same two characters :smile:

**Review by cole-h — APPROVED** [review]
LGTM, and builds fine on proper aarch64 hardware.

Thanks!

**Pull request merged**